### PR TITLE
Rewrite Java Time Libs

### PR DESCRIPTION
**Pull request by chipkent**
* Do a clean reorganization and rewrite of the Deephaven Java time libraries.
* Remove legacy Deephaven types, such as `Period` and `TimeZone`.

### CODE REVIEW

File: engine/time/src/main/java/io/deephaven/time/DateTimeUtils.java [hunk 1]
```diff
@@ -135,19 +135,20 @@ public class DateTimeUtils {
...
      */
     public static final long WEEK = 7 * DAY;
 
     /**
-     * One year (365 days) in nanoseconds.
+     * One average year in nanoseconds. This is 365.2425 days of wall time and does not take into account calendar
+     * adjustments.
      */
-    public static final long YEAR = 365 * DAY;
+    public static final long YEAR = 31556952000000000L;
```
Comment (rbasralian) [inline engine/time/src/main/java/io/deephaven/time/DateTimeUtils.java]: could you define this as `(long) (365.2425 * DAY)`? it's just faster/more clear when looking at the source (especially if you don't have the javadocs available and can't see the comment)

**Review by jmao-denver — CHANGES_REQUESTED** [review]

File: engine/time/src/main/java/io/deephaven/time/DateTimeUtils.java (hunk 1 above)
Comment (chipkent) [reply engine/time/src/main/java/io/deephaven/time/DateTimeUtils.java]: The docs and current revision should make this clear.

**Comment by rcaudy** [comment]
Closes #3900

**Review by jmao-denver — APPROVED** [review]

**Review by jmao-denver — APPROVED** [review]

**Comment by deephaven-internal** [comment]
Labels indicate documentation is required. Issues for documentation have been opened:

How-to: https://github.com/deephaven/deephaven.io/issues/2701
Conceptual: https://github.com/deephaven/deephaven.io/issues/2704
Reference: https://github.com/deephaven/deephaven.io/issues/2702
Blog: https://github.com/deephaven/deephaven.io/issues/2703

**Pull request merged**